skbio/stats/composition.py

```diff
@@ -2058,3 +2059,91 @@ def dirmult_ttest(
         "Reject null hypothesis",
     ]
     return res[col_order]
+
+
+def dirmult_lme(
+    formula, data, groups, reml=True, method=None, fit_kwargs=None, **kwargs
+):
+    r"""Fit a Dirichlet Multinomial linear mixed effects model.
+
+    The Dirichlet-multinomial distribution is a compound distribution that
+    combines a Dirichlet distribution over the probabilities of a multinomial
+    distribution. This distribution is used to model the distribution of
+    species abundances in a community.
+
+    To perform
+
+    This function uses ``MixedLM`` module from ``statsmodels.formula.api``
+
+    Parameters
+    ----------
+    formula : str or generic Formula object
+        The formula specifying the model
+    data : array_like
+        The data for the model. data must define __getitem__ with the keys in
+        the formula terms args and kwargs are passed on to the model
+        instantiation. E.g., a numpy structured or rec array, a dictionary,
+        or a pandas DataFrame.
+    groups : str
+        The column name in data that identifies the grouping variable
+    reml : bool
+        If true, fit according to the REML likelihood, else fit the standard
+        likelihood using ML.
+    method : str
+        Optimization method. Can be a scipy.optimize method name, or a list of such
+        names to be tried in sequence.
+    fit_kwargs : dict
+        Keyword arguments to pass to the model fit function
+    **kwargs
+        Additional keyword arguments to pass to the mixedlm function
+
+    Returns
+    -------
+    results : statsmodels.regression.mixed_linear_model.MixedLMResults
+        The fitted model results.
+        See ``statsmodels.regression.mixed_linear_model.MixedLMResults``
+
+    See Also
+    --------
+    statsmodels.formula.api.mixedlm
+    statsmodels.regression.mixed_linear_model.MixedLM
+
+    Notes
+    -----
+    pass
+
+    Examples
+    --------
+    >>> data = pd.DataFrame([[1, 0, 10.2], [1, 1, 12.5],
+    ...                     [1, 7, 14.8], [1, 14, 16.1]],
+    ...                     index=[0, 1, 2, 3],
+    ...                     columns=["patient_id", "time_point",
+    ...                             "response"])
+    >>> result = dirmult_lme("response ~ time_point", groups="patient_id", data=data)
+    >>> print(result.summary())
+              Mixed Linear Model Regression Results
+    =========================================================
+    Model:             MixedLM  Dependent Variable:  response
+    No. Observations:  4        Method:              REML
```

Comment:
The trailing whitespace is automatically inserted by the `result.summary()`  function but is removed by ruff, which is causing a failure in doctests. I will find other outputs to replace this one.

Reply:
A quick search found [this](https://github.com/astral-sh/ruff/issues/10275). There seems to have no easy solution now. A quick pack would be adding `# doctest: +SKIP` to the end of `print(result.summary())`. This will instruct Sphinx to skip this doctest, regardless the output is correct or not.

Reply:
Thank you so much for your suggestion! I have appended it to the line, and it's no longer posing a problem.